packages/amplify_core_plugin_interface/lib/amplify_api_rest_category.dart

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+part of amplify_core_plugin_interface;
+
+class ApiRestCategory {
+  const ApiRestCategory();
+
+  static List<ApiRestPluginInterface> plugins = [];
+
+  bool addPlugin(ApiRestPluginInterface plugin) {
+    plugins.add(plugin);
```

Comment:
should only be added if it's the first plugin, subsequent calls should throw an error like other categories.

Reply:
thanks, those changes are in our merged branch.